edgelet/kube-client/src/client.rs

```diff
@@ -80,32 +72,40 @@ where
         namespace: &str,
         config_map: &api_core::ConfigMap,
     ) -> impl Future<Item = api_core::ConfigMap, Error = Error> {
-        api_core::ConfigMap::create_core_v1_namespaced_config_map(namespace, config_map, None)
-            .map_err(Error::from)
-            .map(|req| {
-                let fut = self.request(req).and_then(|response| match response {
-                    api_core::CreateCoreV1NamespacedConfigMapResponse::Ok(config_map)
-                    | api_core::CreateCoreV1NamespacedConfigMapResponse::Created(config_map)
-                    | api_core::CreateCoreV1NamespacedConfigMapResponse::Accepted(config_map) => {
-                        Ok(config_map)
-                    }
-                    err => {
-                        debug!("Create config map failed with {:#?}", err);
-                        Err(Error::from(ErrorKind::Response))
-                    }
+        let result =
+            api_core::ConfigMap::create_core_v1_namespaced_config_map(namespace, config_map, None)
+                .map_err(Error::from)
+                .map(|req| {
+                    let fut =
+                        self.request(req).and_then(|response| match response {
+                            api_core::CreateCoreV1NamespacedConfigMapResponse::Ok(config_map)
+                            | api_core::CreateCoreV1NamespacedConfigMapResponse::Created(
+                                config_map,
+                            )
+                            | api_core::CreateCoreV1NamespacedConfigMapResponse::Accepted(
+                                config_map,
+                            ) => Ok(config_map),
+                            err => {
+                                debug!("Create config map failed with {:#?}", err);
+                                Err(Error::from(ErrorKind::Response))
+                            }
+                        });
+
+                    Either::A(fut)
```

Comment:
Ah, this is what you were trying to fix. You don't even need a `match` or `Either` here. Just return `fut` from this closure, `.into_future().flatten()` the whole chain and return that.

Reply:
Much better! Learnt what `flatten` does!